src/main/java/com/spikes2212/util/smartmotorcontrollers/TalonFXWrapper.java

```diff
@@ -0,0 +1,124 @@
+        closedLoopConfig.kP = pidSettings.getkP();
+        closedLoopConfig.kI = pidSettings.getkI();
+        closedLoopConfig.kD = pidSettings.getkD();
+        talonFX.getConfigurator().apply(closedLoopConfig);
+    }
+
+    @Override
+    public void configureFF(FeedForwardSettings feedForwardSettings) {
+        closedLoopConfig.kS = feedForwardSettings.getkS();
+        closedLoopConfig.kV = feedForwardSettings.getkV();
+        closedLoopConfig.kA = feedForwardSettings.getkA();
+        closedLoopConfig.kG = feedForwardSettings.getkG();
+        talonFX.getConfigurator().apply(closedLoopConfig);
+    }
+
+    @Override
+    public void configureTrapezoid(TrapezoidProfileSettings trapezoidProfileSettings) {
+        MotionMagicConfigs config = new MotionMagicConfigs();
+        config.MotionMagicAcceleration = trapezoidProfileSettings.getAccelerationRate();
+        config.MotionMagicCruiseVelocity = trapezoidProfileSettings.getMaxVelocity();
+        config.MotionMagicJerk = trapezoidProfileSettings.getCurve();
+        talonFX.getConfigurator().apply(config);
+    }
+
+    @Override
+    public void pidSet(UnifiedControlMode controlMode, double setpoint, PIDSettings pidSettings,
+                       FeedForwardSettings feedForwardSettings, TrapezoidProfileSettings trapezoidProfileSettings) {
+        configurePID(pidSettings);
+        configureFF(feedForwardSettings);
+        configureTrapezoid(trapezoidProfileSettings);
+        ControlRequest request = switch (controlMode) {
+            case CURRENT -> new TorqueCurrentFOC(setpoint);
+            case PERCENT_OUTPUT -> new DutyCycleOut(setpoint);
+            case TRAPEZOID_PROFILE -> new MotionMagicDutyCycle(setpoint);
+            case MOTION_PROFILING -> throw new UnsupportedOperationException("Motion Profiling is not yet implemented in SpikesLib2!");
+            case VOLTAGE -> new VoltageOut(setpoint);
+            case VELOCITY -> new VelocityDutyCycle(setpoint);
+            case POSITION -> new PositionDutyCycle(setpoint);
+        };
+        talonFX.setControl(request);
+    }
+
+    public void follow(TalonFX master) {
+        talonFX.setControl(new Follower(master.getDeviceID(), false));
+    }
+
+    @Override
+    public void set(double speed) {
+        talonFX.set(speed);
+    }
+
+    @Override
+    public double get() {
+        return talonFX.get();
+    }
+
+    @Override
+    public void setInverted(boolean inverted) {
+        talonFX.getConfigurator().apply(motorOutputConfigs.withInverted(inverted ?
+                InvertedValue.Clockwise_Positive : InvertedValue.CounterClockwise_Positive));
+    }
+
+    @Override
+    public boolean getInverted() {
+        return motorOutputConfigs.Inverted == InvertedValue.Clockwise_Positive;
+    }
+
+    @Override
+    public void disable() {
+        talonFX.disable();
+    }
+
+    @Override
+    public void stopMotor() {
+        talonFX.stopMotor();
+    }
```

Comment:
consider going through the list of functions in TalonFX and think what do u want to expose.
things like:
`getPosition` `getAcceleration` `getMotorVoltage` `setVoltage` etc.